### PR TITLE
Added template parameters mapping to contracts and closures duplicator.

### PR DESCRIPTION
**Pull request by hubuk**
Fix for #380.

### CODE REVIEW

File: Foxtrot/Foxtrot/DuplicatorForContractsAndClosures.cs [hunk 1]
```diff
@@ -72,6 +72,15 @@ internal class DuplicatorForContractsAndClosures : Duplicator
                         dup.DuplicateFor[sourceMethod.Parameters[i].UniqueKey] = targetMethod.Parameters[i];
                     }
                 }
+
+                if (sourceMethod.TemplateParameters != null && targetMethod.TemplateParameters != null
```
Comment (SergeyTeplyakov) [inline Foxtrot/Foxtrot/DuplicatorForContractsAndClosures.cs]: The only suggestion is to put a comment that will explain (very shortly) what problem this code is trying to solve.

**Comment by SergeyTeplyakov** [comment]
@hubuk Great job!!! Put one comment with minor suggestion.

**Comment by hubuk** [comment]
@SergeyTeplyakov Done, let me know if there is anything I should add to the comment.

**Comment by SergeyTeplyakov** [comment]
@hubuk, excellent. Once check finishes, I'll merge this change.

**Pull request merged**